sql/snapshot.sql

```diff
@@ -0,0 +1,546 @@
+)
+INHERITS (public.abstract_transactions);
+
+-- Table Triggers
+
+create trigger set_updated_at_fee_transaction before
+update
+    on
+    public.fee_transactions for each row execute function update_updated_at_column();
+
+
+-- public.metagraph_balance_changes definition
+
+-- Drop table
+
+-- DROP TABLE metagraph_balance_changes;
+
+CREATE TABLE metagraph_balance_changes (
+	metagraph_id varchar NOT NULL,
+	metagraph_hash varchar NOT NULL,
+	address varchar NOT NULL,
+	balance int8 NOT NULL,
+	created_at timestamp DEFAULT now() NOT NULL,
+	updated_at timestamp DEFAULT now() NOT NULL,
+	metagraph_snapshot_ordinal int8 NOT NULL,
+	CONSTRAINT metagraph_balance_change_pk PRIMARY KEY (metagraph_id, address, metagraph_snapshot_ordinal),
+	CONSTRAINT metagraph_balance_change_metagraph_snapshot_fk FOREIGN KEY (metagraph_id,metagraph_hash) REFERENCES metagraph_snapshots(metagraph_id,hash)
+);
+
+-- Table Triggers
+
+create trigger set_updated_at_metagraph_balance_change before
+update
+    on
+    public.metagraph_balance_changes for each row execute function update_updated_at_column();
+
+
+-- public.metagraph_blocks definition
+
+-- Drop table
+
+-- DROP TABLE metagraph_blocks;
+
+CREATE TABLE metagraph_blocks (
+	metagraph_id varchar NOT NULL,
+	snapshot_hash varchar NOT NULL,
+	CONSTRAINT metagraph_block_pk PRIMARY KEY (metagraph_id, hash),
+	CONSTRAINT metagraph_block_metagraph_snapshot_fk FOREIGN KEY (metagraph_id,snapshot_hash) REFERENCES metagraph_snapshots(metagraph_id,hash)
+)
+INHERITS (public.abstract_blocks);
+
+-- Table Triggers
+
+create trigger set_updated_at_metagraph_blocks before
+update
+    on
+    public.metagraph_blocks for each row execute function update_updated_at_column();
+
+
+-- public.metagraph_reward_transactions definition
+
+-- Drop table
+
+-- DROP TABLE metagraph_reward_transactions;
+
+CREATE TABLE metagraph_reward_transactions (
+	metagraph_id varchar NOT NULL,
+	metagraph_snapshot_hash varchar NOT NULL,
+	CONSTRAINT metagraph_reward_transaction_pk PRIMARY KEY (hash),
+	CONSTRAINT metagraph_reward_transaction_metagraph_reward_transaction_fk FOREIGN KEY (metagraph_id,metagraph_snapshot_hash) REFERENCES metagraph_snapshots(metagraph_id,hash)
+)
+INHERITS (public.abstract_transactions);
+
+-- Table Triggers
+
+create trigger set_updated_at_metagraph_reward_transaction before
+update
+    on
+    public.metagraph_reward_transactions for each row execute function update_updated_at_column();
+
+
+-- public.metagraph_transactions definition
+
+-- Drop table
+
+-- DROP TABLE metagraph_transactions;
+
+CREATE TABLE metagraph_transactions (
+	metagraph_id varchar NOT NULL,
+	block_hash varchar NOT NULL,
+	CONSTRAINT destination_addr_not_null CHECK ((destination_addr IS NOT NULL)),
+	CONSTRAINT metagraph_transaction_pk PRIMARY KEY (hash),
+	CONSTRAINT original_not_null CHECK ((original IS NOT NULL)),
```

Comment:
We don't want to save an original transaction for anything except dag transactions (for backwards compatibility). Otherwise it's just non-normalized data.